x-pack/legacy/plugins/siem/public/components/fields_browser/field_browser.tsx

```diff
@@ -101,34 +101,79 @@ type Props = Pick<
  * This component has no internal state, but it uses lifecycle methods to
  * set focus to the search input, scroll to the selected category, etc
  */
-export class FieldsBrowser extends React.PureComponent<Props> {
-  public componentDidMount() {
-    this.scrollViews();
-    this.focusInput();
-  }
+export const FieldsBrowser = React.memo<Props>(
+  ({
+    browserFields,
+    columnHeaders,
+    filteredBrowserFields,
+    isEventViewer,
+    isSearching,
+    onCategorySelected,
+    onFieldSelected,
+    onHideFieldBrowser,
+    onSearchInputChange,
+    onOutsideClick,
+    onUpdateColumns,
+    searchInput,
+    selectedCategoryId,
+    timelineId,
+    toggleColumn,
+    width,
+  }) => {
+    /** Focuses the input that filters the field browser */
+    const focusInput = () => {
+      const elements = document.getElementsByClassName(
+        getFieldBrowserSearchInputClassName(timelineId)
+      );
 
-  public componentDidUpdate() {
-    this.scrollViews();
-    this.focusInput(); // always re-focus the input to enable additional filtering
-  }
+      if (elements.length > 0) {
+        (elements[0] as HTMLElement).focus(); // this cast is required because focus() does not exist on every `Element` returned by `getElementsByClassName`
+      }
+    };
 
-  public render() {
-    const {
-      columnHeaders,
-      browserFields,
-      filteredBrowserFields,
-      searchInput,
-      isEventViewer,
-      isSearching,
-      onCategorySelected,
-      onFieldSelected,
-      onOutsideClick,
-      onUpdateColumns,
-      selectedCategoryId,
-      timelineId,
-      toggleColumn,
-      width,
-    } = this.props;
+    /** Invoked when the user types in the input to filter the field browser */
+    const onInputChange = (event: React.ChangeEvent<HTMLInputElement>) =>
+      onSearchInputChange(event.target.value);
+
+    const selectFieldAndHide: OnFieldSelected = (fieldId: string) => {
+      if (onFieldSelected != null) {
+        onFieldSelected(fieldId);
+      }
+
+      onHideFieldBrowser();
+    };
+
+    const scrollViews = () => {
+      if (selectedCategoryId !== '') {
+        const categoryPaneTitles = document.getElementsByClassName(
+          getCategoryPaneCategoryClassName({
+            categoryId: selectedCategoryId,
+            timelineId,
+          })
+        );
+
+        if (categoryPaneTitles.length > 0) {
+          categoryPaneTitles[0].scrollIntoView();
+        }
+
+        const fieldPaneTitles = document.getElementsByClassName(
+          getFieldBrowserCategoryTitleClassName({
+            categoryId: selectedCategoryId,
+            timelineId,
+          })
+        );
+
+        if (fieldPaneTitles.length > 0) {
+          fieldPaneTitles[0].scrollIntoView();
+        }
+      }
+
+      focusInput(); // always re-focus the input to enable additional filtering
+    };
+    useEffect(() => {
```

Comment:
I am wondering if we can just do that,  I removed the `focusInput` function in the `useEffect` since we are already doing it in the function scrollViews. I also think that we can these dependencies to make it more robust.
```
useEffect(() => {
      scrollViews();
    }, [selectedCategoryId, timelineId]);
```

Reply:
ah, yes i meant to come back to this, was waiting on Andrew's response on slack! good catch